server/isomorphic-routing/index.js

```diff
@@ -40,13 +50,18 @@ function getEnhancedContext( req, res ) {
 	} );
 }
 
-function applyMiddlewares( context, ...middlewares ) {
-	const liftedMiddlewares = middlewares.map( middleware => next => middleware( context, next ) );
+function applyMiddlewares( context, expressNext, ...middlewares ) {
+	const liftedMiddlewares = middlewares.map( middleware => next => middleware( context, ( err ) => {
+		if ( err ) {
+			expressNext( err ); // Call express' next( err ) for error handling (and bail early from this route)
```

Comment:
Yeah, this whole file is a bit magical. Essentially, it's an adapter to make `page.js` middleware (signature: `context, next`) work with Express (middleware signature: `req, res, next`). Or, for [error handling](http://expressjs.com/en/guide/error-handling.html), `err, req, res, next`.

We pass in Express' `next` as `expressNext` here, while we construct an artificial Express middleware by stacking together all page.js-style middlewares we're given (hence the funny `middleware => next => ...` construct). If `expressNext` is invoked with an arg, that's interpreted as an error object, and the next error handling mw is invoked. We need to make sure this works with our adapter, too, so this is essentially what's going on here.